File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        int power = number.ToString().Length;
        int currentValue = number;
        long sum = 0;

        while (currentValue > 0)
        {
            sum += (long)Math.Pow(currentValue % 10, power);
            currentValue /= 10;
        }

        return number == sum;
    }
}

File: go/armstrong-numbers/armstrong_numbers.go
package armstrong

import (
	"math"
	"strconv"
)

func IsNumber(n int) bool {

	power := float64(len(strconv.Itoa(n)))
	current := n
	var sum float64

	for current > 0 {
		t := float64(current % 10)
		sum += math.Pow(t, power)
		current /= 10
	}

	return float64(n) == sum
}
